linera-service/src/linera/main.rs

```diff
@@ -1448,6 +1459,110 @@ impl ClientOptions {
+        S: KeyValueStore + Clone + Send + Sync + 'static,
+        S::Error: Send + Sync,
+    {
+        match self.0 {
+            DatabaseToolCommand::DeleteAll => {
+                let start_time = Instant::now();
+                S::delete_all(&config).await?;
+                info!(
+                    "All namespaces deleted in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+            }
+            DatabaseToolCommand::DeleteNamespace => {
+                let start_time = Instant::now();
+                S::delete(&config, &namespace).await?;
+                info!(
+                    "Namespace deleted in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+            }
+            DatabaseToolCommand::CheckExistence => {
+                let start_time = Instant::now();
+                let test = S::exists(&config, &namespace).await?;
+                info!(
+                    "Existence of a namespace checked in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+                if test {
+                    println!("The database does exist");
+                    return Ok(0);
+                } else {
+                    println!("The database does not exist");
+                    return Ok(1);
+                }
+            }
+            DatabaseToolCommand::CheckAbsence => {
+                let start_time = Instant::now();
+                let test = S::exists(&config, &namespace).await?;
+                info!(
+                    "Absence of a namespace checked in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+                if test {
+                    println!("The database does exist");
+                    return Ok(1);
+                } else {
+                    println!("The database does not exist");
+                    return Ok(0);
+                }
+            }
+            DatabaseToolCommand::Initialize {
+                genesis_config_path,
+            } => {
+                let start_time = Instant::now();
+                let genesis_config: GenesisConfig = util::read_json(genesis_config_path)?;
+                let mut storage =
+                    DbStorage::<S, _>::maybe_create_and_connect(&config, &namespace, None).await?;
+                genesis_config.initialize_storage(&mut storage).await?;
+                info!(
+                    "Initialization done in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+            }
+            DatabaseToolCommand::ListNamespaces => {
+                let start_time = Instant::now();
+                let namespaces = S::list_all(&config).await?;
+                info!(
+                    "Namespaces listed in {} ms",
+                    start_time.elapsed().as_millis()
+                );
+                println!("The list of namespaces is {:?}", namespaces);
```

Comment:
In all three `List*` commands, too: Should the `println!` output be machine-readable?